configurator/src/main.rs

```diff
@@ -280,6 +340,28 @@ fn main() -> Result<(), anyhow::Error> {
         }
     }
 
+    let mut macaroon_vec: Vec<u8> = Vec::new();
+
+    if config.advanced.plugins.rest {
+        let macaroon_path = Path::new(
+            "/usr/local/libexec/c-lightning/plugins/c-lightning-REST/certs/access.macaroon",
+        );
+        while !macaroon_path.exists() {
+            std::thread::sleep(std::time::Duration::from_secs(1));
+        }
+        std::fs::create_dir_all("/root/.lightning/public")?;
```

Comment:
This will create the directory while we're waiting for the file to be written, taking advantage of concurrency better.

```suggestion
        std::fs::create_dir_all("/root/.lightning/public")?;
        while !macaroon_path.exists() {
            std::thread::sleep(std::time::Duration::from_secs(1));
        }
```